src/user/user.repository.ts

```diff
@@ -48,4 +47,40 @@ export class UserRepository {
       where: { id },
     });
   }
+
+  async createPat(userId: number, token: string) {
+    const test = this.prisma.pAT.create({
+      data: {
+        token,
+        user: {
+          connect: {
+            id: userId,
+          },
+        },
+      },
+    });
+    return test;
+  }
+
+  async getUserValidPatWithId(userId: number): Promise<PAT> {
+    return this.prisma.pAT.findFirst({
+      where: {
+        userId,
+        invalidatedAt: {
+          equals: null,
+        },
+      },
+    });
+  }
+
+  async invalidateUserPat(userId: number): Promise<void> {
+    await this.prisma.pAT.updateMany({
```

Comment:
you can remove `async` and `await` and keep `Promise<void>` - you will still be able to `await` it where you call this method

Reply:
I've tested it out and without `async` `await` in `user.repository` it doesn't update `invalidatedAt` field (I awaited this in `auth.service`).

Reply:
hmm, have you tried returning it? I mean adding `return` statement.

However, we might not want to return anything (so leaving `Promise<void>` intact) -in this case what you have now is correct 👍🏻